perception/autoware_tensorrt_bevdet/src/bevdet.cpp

```diff
@@ -0,0 +1,711 @@
+
+    size_t buf_size = trt_buffer_sizes[buffer_map["adj_feats"]] / adj_num;
+
+    CHECK_CUDA(cudaMemcpy(
+      reinterpret_cast<char *>(trt_buffer_dev[buffer_map["adj_feats"]]) + i * buf_size, adj_buffer,
+      buf_size, cudaMemcpyDeviceToDevice));
+
+    Eigen::Quaternion<float> adj_ego2global_rot;
+    Eigen::Translation3f adj_ego2global_trans;
+    adj_frame_ptr->getEgo2Global(i, adj_ego2global_rot, adj_ego2global_trans);
+
+    GetCurr2AdjTransform(
+      ego2global_rot, adj_ego2global_rot, ego2global_trans, adj_ego2global_trans,
+      reinterpret_cast<float *>(trt_buffer_dev[buffer_map["transforms"]]) + i * transform_size);
+  }
+  CHECK_CUDA(cudaMemcpy(
+    trt_buffer_dev[buffer_map["flag"]], &flag, trt_buffer_sizes[buffer_map["flag"]],
+    cudaMemcpyHostToDevice));
+}
+
+void BEVDet::GetCurr2AdjTransform(
+  const Eigen::Quaternion<float> & curr_ego2global_rot,
+  const Eigen::Quaternion<float> & adj_ego2global_rot,
+  const Eigen::Translation3f & curr_ego2global_trans,
+  const Eigen::Translation3f & adj_ego2global_trans, float * transform_dev)
+{
+  Eigen::Matrix4f curr_e2g_transform;
+  Eigen::Matrix4f adj_e2g_transform;
+
+  for (int i = 0; i < 3; i++) {
+    for (int j = 0; j < 3; j++) {
+      curr_e2g_transform(i, j) = curr_ego2global_rot.matrix()(i, j);
+      adj_e2g_transform(i, j) = adj_ego2global_rot.matrix()(i, j);
+    }
+  }
+  for (int i = 0; i < 3; i++) {
+    curr_e2g_transform(i, 3) = curr_ego2global_trans.vector()(i);
+    adj_e2g_transform(i, 3) = adj_ego2global_trans.vector()(i);
+
+    curr_e2g_transform(3, i) = 0.f;
+    adj_e2g_transform(3, i) = 0.f;
+  }
+  curr_e2g_transform(3, 3) = 1.f;
+  adj_e2g_transform(3, 3) = 1.f;
+
+  Eigen::Matrix4f currEgo2adjEgo = adj_e2g_transform.inverse() * curr_e2g_transform;
+  Eigen::Matrix3f currEgo2adjEgo_2d;
+  for (int i = 0; i < 2; i++) {
+    for (int j = 0; j < 2; j++) {
+      currEgo2adjEgo_2d(i, j) = currEgo2adjEgo(i, j);
+    }
+  }
+  currEgo2adjEgo_2d(2, 0) = 0.f;
+  currEgo2adjEgo_2d(2, 1) = 0.f;
+  currEgo2adjEgo_2d(2, 2) = 1.f;
+  currEgo2adjEgo_2d(0, 2) = currEgo2adjEgo(0, 3);
+  currEgo2adjEgo_2d(1, 2) = currEgo2adjEgo(1, 3);
+
+  Eigen::Matrix3f gridbev2egobev;
+  gridbev2egobev(0, 0) = x_step;
+  gridbev2egobev(1, 1) = y_step;
+  gridbev2egobev(0, 2) = x_start;
+  gridbev2egobev(1, 2) = y_start;
+  gridbev2egobev(2, 2) = 1.f;
+
+  gridbev2egobev(0, 1) = 0.f;
+  gridbev2egobev(1, 0) = 0.f;
+  gridbev2egobev(2, 0) = 0.f;
+  gridbev2egobev(2, 1) = 0.f;
+
+  Eigen::Matrix3f currgrid2adjgrid = gridbev2egobev.inverse() * currEgo2adjEgo_2d * gridbev2egobev;
+
+  CHECK_CUDA(cudaMemcpy(
+    transform_dev, Eigen::Matrix3f(currgrid2adjgrid.transpose()).data(),
+    transform_size * sizeof(float), cudaMemcpyHostToDevice));
+}
+
+int BEVDet::DoInfer(
+  const camsData & cam_data, std::vector<Box> & out_detections, float & cost_time, int idx)
+{
+  printf("-------------------%d-------------------\n", idx + 1);
```

Comment:
@cyn-liu 

Please use `RCLCPP_INFO` :pray:

Reply:
In theory, these printed information:
```
printf("-------------------%d-------------------\n", idx + 1);
printf("TRT-Engine  : %.5lf ms\n", engine_t.count() * 1000);
printf("Postprocess : %.5lf ms\n", post_t.count() * 1000);
printf("Inference   : %.5lf ms\n", infer_t.count() * 1000);

printf("Detect %ld objects\n", out_detections.size());
```
should be deleted, just so that when running the `tensorrt_bevdet_node` separately, some detection information can be seen, so there is no need to change it to `RCLCPP_INFO`.

At present, no one should be able to directly use the output of `tensorrt_bevdet_node` for Autoware autonomous driving testing, so these printed information can help debug this node separately. When this node matures, these printed information should be deleted.

Reply:
> When this node matures, these printed information should be deleted

Can you use `RCLCPP_DEBUG` for this kind of log?
If you use `printf`, it is difficult for user to suppress the debug information when the user runs autoware with bevdet.

Reply:
I second @Shin-kyoto on this. We do not want to contaminate the logs when launch Autoware in general and using RCLCPP_INFO, RCLCPP_DEBUG, RCLCPP_ERROR helps us select the verbosity of terminal output depending on the user's needs.